packages/sdk-node/src/authenticateRequest.ts

```diff
@@ -104,22 +104,20 @@ const isDevelopmentFromApiKey = (apiKey: string): boolean =>
   apiKey.startsWith('test_') || apiKey.startsWith('sk_test_');
 
 const getRequestUrl = (req: IncomingMessage): URL => {
-  return new URL(req.url as string, `${getRequestProto(req)}://${req.headers.host}`);
-};
-
-const getRequestProto = (req: IncomingMessage): string => {
   // @ts-ignore Optimistic attempt to get the protocol in case
   // req extends IncomingMessage in a useful way. No guarantee
   // it'll work.
-  const mightWork = req.connection?.encrypted ? 'https' : 'http';
-  // The x-forwarded-proto header takes precedence.
-  const proto = (req.headers[constants.Headers.ForwardedProto] as string) || mightWork;
-  if (!proto) {
+  const protocol = req.connection?.encrypted ? 'https' : 'http';
+  const forwardedProto = req.headers[constants.Headers.ForwardedProto] as string;
+  const host = req.headers.host;
+  const forwardedHost = req.headers[constants.Headers.ForwardedHost] as string;
+
+  const origin = buildOrigin({ protocol, forwardedProto, forwardedHost, host });
+  if (!origin) {
```

Comment:
❓ If the origin is falsy, it means it's missing proto? Because that's what the Error here indicates.

Reply:
you are correct. The error does not make much sense after refactoring. Should i replace it to another error? Any suggestion?

Reply:
@dimkl What are the reasons that `buildOrigin` may return a falsy value?

Reply:
when `protocol && forwardedProto` is falsy or `forwardedHost && host` is falsy.